packages/native/src/useLinking.native.tsx

```diff
@@ -134,6 +135,17 @@ export function useLinking(
     []
   );
 
+  const validateRoutesNotExistInRootState = React.useCallback(
```

Comment:
Avoid `useCallback` for these cases as React doesn't guarantee that it won't be cleared. Use `use-latest-callback` instead. or better, define this function inside the effect/inline the logic as before.